From c5754981cb6feb7e2ed11193b819f9dc788facdd Mon Sep 17 00:00:00 2001
From: "Philip, Avinash" <avinashphilip@ti.com>
Date: Tue, 6 Mar 2012 09:26:51 +0530
Subject: [PATCH 073/193] Revert "arm:omap:nand - Synching HAMMING ECC layout
 with that of U-boot."

This reverts commit b60fadc071d76154972513623cf7e784a3af1d00.
---
 drivers/mtd/nand/omap2.c |    9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/drivers/mtd/nand/omap2.c b/drivers/mtd/nand/omap2.c
index 4e5073b..34423dd 100644
--- a/drivers/mtd/nand/omap2.c
+++ b/drivers/mtd/nand/omap2.c
@@ -1156,14 +1156,7 @@ static int __devinit omap_nand_probe(struct platform_device *pdev)
 			omap_oobinfo.oobfree->offset = offset;
 			omap_oobinfo.oobfree->length = info->mtd.oobsize -
 						offset - omap_oobinfo.eccbytes;
-			/*
-			offset is calculated considering the following :
-			1) 12 bytes ECC for 512 byte access and 24 bytes ECC for
-			256 byte access in OOB_64 can be supported
-			2)Ecc bytes lie to the end of OOB area.
-			3)Ecc layout must match with u-boot's ECC layout.
-			*/
-			offset = info->mtd.oobsize - MAX_HWECC_BYTES_OOB_64;
+			offset = info->mtd.oobsize - omap_oobinfo.eccbytes;
 		}
 
 		for (i = 0; i < omap_oobinfo.eccbytes; i++)
-- 
1.7.9.5

